src/common/njord.ts

```diff
@@ -644,6 +644,53 @@ export const awardPost = async (
           )
           .execute();
 
+        let newComment: Comment | undefined;
+
+        if (note) {
+          newComment = entityManager.getRepository(Comment).create({
+            id: await generateShortId(),
+            postId: post.id,
+            parentId: null,
+            userId: ctx.userId,
+            content: note,
+            awardTransactionId: transaction.id,
+            flags: {
+              awardId: transaction.productId,
+            },
+          });
+
+          newComment.flags = {
+            ...newComment.flags,
+            vordr: await checkWithVordr(
+              {
+                id: newComment.id,
+                type: VordrFilterType.Comment,
+                content: newComment.content,
+              },
+              ctx,
+            ),
+          };
+
+          await saveComment(entityManager, newComment, post.sourceId);
+          await entityManager
+            .getRepository(UserComment)
```

Comment:
I don't think we need UserComment here because it would imply user awarded their own comment. We already award post through `UserPost`.

Reply:
Oh wasn't sure what UserComment was again, thought just the link between comment-user
Why we need it on comments then?